src/gui/DetailsWidget.cpp

```diff
@@ -0,0 +1,321 @@
+
+    m_ui->totpButton->hide();
+    m_ui->totpWidget->hide();
+    m_ui->totpButton->setChecked(false);
+
+    auto icon = m_currentEntry->iconPixmap();
+    if (icon.width() > 16 || icon.height() > 16) {
+        icon = icon.scaled(16, 16);
+    }
+    m_ui->entryIcon->setPixmap(icon);
+
+    QString title = QString(" / ");
+    Group* entry_group = m_currentEntry->group();
+    if (entry_group) {
+        QStringList hierarchy = entry_group->hierarchy();
+        hierarchy.removeFirst();
+        title += hierarchy.join(" / ");
+        if (hierarchy.size() > 0) {
+            title += " / ";
+        }
+    }
+    title.append(m_currentEntry->resolveMultiplePlaceholders(m_currentEntry->title()));
+    m_ui->titleLabel->setText(title);
+
+    m_ui->usernameLabel->setText(m_currentEntry->resolveMultiplePlaceholders(m_currentEntry->username()));
+
+    if (!config()->get("security/hidepassworddetails").toBool()) {
+        m_ui->passwordLabel->setText(shortPassword(m_currentEntry->resolveMultiplePlaceholders(m_currentEntry->password())));
+        m_ui->passwordLabel->setToolTip(m_currentEntry->resolveMultiplePlaceholders(m_currentEntry->password()));
+    } else {
+        m_ui->passwordLabel->setText("****");
+    }
+
+    QString url = m_currentEntry->webUrl();
+    if (url != "") {
+        url = QString("<a href=\"%1\">%2</a>").arg(url).arg(shortUrl(url));
+        m_ui->urlLabel->setOpenExternalLinks(true);
+    } else {
+        url = shortUrl(m_currentEntry->resolveMultiplePlaceholders(m_currentEntry->url()));
+        m_ui->urlLabel->setOpenExternalLinks(false);
+    }
+    m_ui->urlLabel->setText(url);
+
+    TimeInfo entryTime = m_currentEntry->timeInfo();
+    if (entryTime.expires()) {
+        m_ui->expirationLabel->setText(entryTime.expiryTime().toString(Qt::DefaultLocaleShortDate));
+    } else {
+        m_ui->expirationLabel->setText(tr("Never"));
+    }
+
+    if (m_currentEntry->hasTotp()) {
+        m_ui->totpButton->show();
+        updateTotp();
+
+        m_step = m_currentEntry->totpStep();
+
+        m_timer = new QTimer(this);
+        connect(m_timer, SIGNAL(timeout()), this, SLOT(updateTotp()));
+        m_timer->start(m_step * 10);
+    }
+
+    QString notes = m_currentEntry->notes();
+    if (!notes.isEmpty()) {
+        m_ui->tabWidget->setTabEnabled(NotesTab, true);
+        m_ui->notesEdit->setText(m_currentEntry->resolveMultiplePlaceholders(notes));
+    }
+
+    QStringList customAttributes = m_currentEntry->attributes()->customKeys();
+    if (customAttributes.size() > 0) {
+        m_ui->tabWidget->setTabEnabled(AttributesTab, true);
+        m_ui->attributesEdit->clear();
+
+        QString attributesText = QString();
+        for (const QString& key : customAttributes) {
+            QString value = m_currentEntry->attributes()->value(key);
+            if (m_currentEntry->attributes()->isProtected(key)) {
+                value = "<i>[PROTECTED]</i>";
```

Comment:
Not translated. Re-use locale string from edit entry screen (without the `i` tags).

Reply:
In the edit entry it's `tr("[PROTECTED] Press reveal to view or edit")`

I will separate the 2 string so we can re-use `[PROTECTED]`